frontend/src/hooks/useDebounce.js

```diff
@@ -0,0 +1,26 @@
+// source: https://usehooks.com/useDebounce/
+import { useEffect, useState } from "react";
+
+export const useDebounce = (value, delay) => {
```

Comment:
instead of something like this i usually like something like https://www.npmjs.com/package/debounce-promise
but if you already have lodash as a dependancy, its not worth it and sticking to this is a good idea.